src/components/Search.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import EmployeeList from "./EmployeeList";
+import { EmployeeInterface } from "./Employee";
+import Loading from "./Loading";
+import fetchEmployees from "../utils/fetchEmployees";
+
+const Search = (): JSX.Element => {
+  const [matchedEmployees, setMatchedEmployees] = useState<EmployeeInterface[]>(
+    []
+  );
+  const [loading, setLoading] = useState<boolean>(true);
+  const [searchParams] = useSearchParams();
+  let query: string | null = searchParams.get("query");
+
+  if (query) {
+    query = query.toLowerCase();
+  }
+
+  useEffect(() => {
+    fetchEmployees(
```

Comment:
You shouldn't fetch employees multiple times. Fetch them once and store them somewhere. Then, operate on them.

Reply:
I think setting up a React context for employees would make things easier.